nesi_jupyter_helpers/add_kernel.py

```diff
@@ -21,6 +21,9 @@
 # load required modules
 module load slurm NeSI  # ensure these modules gets loaded even on Maui ancil.
 {modules_txt}
+
+# isolate conda/virtual environments from user's site-packages directory
+export PYTHONNOUSERSITE=True
```

Comment:
@jennan I'm not sure we should force this for all kernels (forcing non-default behaviour could be confusing in some cases). It probably does make sense to have it set for conda/virtual environments but then I think it should be moved into those templates instead of the main one. Possibly we could have it as an option so it can be turned on or off by the user and maybe print something about it too?